test-utils/src/gen/controller/pool_state/v2.rs

```diff
@@ -33,6 +36,17 @@ pub fn pool_state_v2_u8_bools_normal_strat() -> PoolStateV2U8Bools<Option<BoxedS
     )
 }
 
+pub fn pool_state_v2_u64s_solvent_strat() -> impl Strategy<Value = PoolStateV2U64s<u64>> {
+    any::<u64>().prop_flat_map(|tsv| {
+        bals_from_supply::<2>(tsv).prop_map(move |([withheld, protocol_fee], _rem)| {
+            PoolStateV2U64s::default()
+                .with_total_sol_value(tsv)
+                .with_withheld_lamports(withheld)
+                .with_protocol_fee_lamports(protocol_fee.min(u64::MAX / 1000))
```

Comment:
I think in most cases we dont want this `u64::MAX / 1000` upper bound. Maybe have `tsv` as an arg thats set by the strategies in `withdraw_protocol_fee` that dont result in overflow when converted to inf?